Tests/PostgresNIOTests/New/Extensions/PSQLBackendMessageEncoder.swift

```diff
@@ -0,0 +1,279 @@
+            buffer.writeBackendMessageID(message.id)
+            self.encodePayload(description, into: &buffer)
+            
+        case .parameterStatus(let status):
+            buffer.writeBackendMessageID(message.id)
+            self.encodePayload(status, into: &buffer)
+            
+        case .readyForQuery(let transactionState):
+            buffer.writeBackendMessageID(message.id)
+            self.encodePayload(transactionState, into: &buffer)
+            
+        case .rowDescription(let description):
+            buffer.writeBackendMessageID(message.id)
+            self.encodePayload(description, into: &buffer)
+            
+        case .sslSupported:
+            buffer.writeInteger(UInt8(ascii: "S"))
+            
+        case .sslUnsupported:
+            buffer.writeInteger(UInt8(ascii: "N"))
+        }
+    }
+    
+    private struct EmptyPayload: PSQLMessagePayloadEncodable {
+        func encode(into buffer: inout ByteBuffer) {}
+    }
+    
+    private struct StringPayload: PSQLMessagePayloadEncodable {
+        var string: String
+        init(_ string: String) { self.string = string }
+        func encode(into buffer: inout ByteBuffer) {
+            buffer.writeNullTerminatedString(self.string)
+        }
+    }
+
+    private func encodePayload<Payload: PSQLMessagePayloadEncodable>(_ payload: Payload, into buffer: inout ByteBuffer) {
+        let startIndex = buffer.writerIndex
+        buffer.writeInteger(Int32(0)) // placeholder for length
+        payload.encode(into: &buffer)
+        let length = Int32(buffer.writerIndex - startIndex)
+        buffer.setInteger(length, at: startIndex)
+    }
+}
+
+extension PSQLBackendMessage {
+    var id: ID {
+        switch self {
+        case .authentication:
+            return .authentication
+        case .backendKeyData:
+            return .backendKeyData
+        case .bindComplete:
+            return .bindComplete
+        case .closeComplete:
+            return .closeComplete
+        case .commandComplete:
+            return .commandComplete
+        case .dataRow:
+            return .dataRow
+        case .emptyQueryResponse:
+            return .emptyQueryResponse
+        case .error:
+            return .error
+        case .noData:
+            return .noData
+        case .notice:
+            return .noticeResponse
+        case .notification:
+            return .notificationResponse
+        case .parameterDescription:
+            return .parameterDescription
+        case .parameterStatus:
+            return .parameterStatus
+        case .parseComplete:
+            return .parseComplete
+        case .portalSuspended:
+            return .portalSuspended
+        case .readyForQuery:
+            return .readyForQuery
+        case .rowDescription:
+            return .rowDescription
+        case .sslSupported,
+             .sslUnsupported:
+            preconditionFailure("Message has no id.")
```

Comment:
What guarantees in practical terms (a runtime crash is not ideal) that no one will ever ask for the `id` of a `ssl[Un]supported` message? It's feeling a lot like these two messages are really not of the same "type" as the rest.

Reply:
It being in a test target :)